### ISSUE TITLE
equivalent of fatal warnings

### ISSUE THREAD

**Issue by fommil** [scalastyle/scalastyle-sbt-plugin#18]
perhaps with the ability to toggle specific style checks on a per-class basis.

this would enable teams with CI to automatically enforce standards in pull requests.

**Comment by matthewfarwell** [comment]
What is the difference between error and fatal?

**Comment by fommil** [comment]
With the setting enabled, the compiler bumps warnings to errors.

**Comment by matthewfarwell** [comment]
Ok, so if I set this flag, and I had a warning, then this would automatically be bumped to an error. Given that when you have errors, and scalastyleFailOnError is set to true, the scalastyle task fails, what more does this give me?

**Comment by fommil** [comment]
selectively bump warnings to errors, so that the build cannot pass if the warts are in the code (but allowing the project to define what constitutes a build failure)

**Comment by ngbinh** [comment]
I would like to have an option to fail the build on warning. Problem with bumping all warning to error on ```scalastyle_config.xml``` is that it makes Idea IntelliJ Scala plugin display all violations as straight error red. Normally, I would like to see it as warning inside IntelliJ.

**Comment by matthewfarwell** [comment]
@ngbinh That is a slightly different issue, and it is easy enough to add. We have very similar functionality in the maven plugin - failOnWarning. Do you want to submit another issue & a PR?

**Comment by ngbinh** [comment]
@matthewfarwell sure. Will look into another issue and PR.
